traffic_ops/app/lib/API/Configs/ApacheTrafficServer.pm

```diff
@@ -2198,6 +2198,7 @@ sub cachegroup_profiles {
 				port           => $self->profile_param_value( $pid, 'parent.config', 'port', undef ),
 				use_ip_address => $self->profile_param_value( $pid, 'parent.config', 'use_ip_address', 0 ),
 				rank           => $self->profile_param_value( $pid, 'parent.config', 'rank', 1 ),
+				not_a_parent   => $self->profile_param_value( $pid, 'parent.config', 'not_a_parent', 'false' ),
```

Comment:
'false' is actually a true value.   To be idiomatic,  you should use undef or 0 for a false value as `use_ip_address` does above.

Reply:
It's a true value in Perl, but "false" isn't a variable in Perl, it's a Parameter in Traffic Ops / Traffic Portal. IMO "false" is the most obvious human-read text we can put in the website.

Reply:
Kinda. "False", "FALSE", "0", and "no" all execute as "true". And the other boolean field in this config is an integer.

Still... this PR is just to bring the existing functionality into the API. It might be worth a separate PR if we want to change the magic values, though.

Reply:
Yes, this PR is simply fixing a bug, and reproducing the existing functionality of #1622.

And for this particular parameter, it defaults to false, and generally shouldn't exist. The only reason it even has as sentinel, is to make it possible to exist and be false, in case someone for some reason wants their parameter to always exist.

And the only reason this particular line is defaulting the human-readable "false" instead of the Perl false `undef`, is so the other line in this PR only has to check one value instead of two.

Reply:
We've already got this in place with 1622.  Any problem with me merging, @dangogh?

Reply:
@alficles not to perl:   ```
$ perl -le "if ('false') { print 'false is true'; }"
false is true
```

to me,  it's confusing when an assumption like this is made.   but,  it's done elsewhere in the code,  so I'm not going to block the change.   @dg4prez go ahead..